sklearn/linear_model/cd_fast.pyx

```diff
@@ -363,7 +418,7 @@ def sparse_enet_coordinate_descent(double[:] w,
             startptr = endptr
 
         # tol *= np.dot(y, y)
-        tol *= ddot(n_samples, <DOUBLE*>&y[0], 1, <DOUBLE*>&y[0], 1)
+        tol *= dot(n_samples, <floating*>&y[0], 1, <floating*>&y[0], 1)
```

Comment:
Yes, but it will appear `Cannot assign type 'char *' to 'float *'`.
So typecasts are still necessary.